### DISCUSSION TITLE
Rekomendasi UUID untuk pengiriman SATUSEHAT

### DISCUSSION THREAD

**Discussion by fattakbar** [ivanwilliammd/satusehat-integration#122] [Q&A (Tanya Jawab)] [answered]
saya mencoba seperti pada dokumentasi untuk melakukan POST Organization dengan code seperti ini
![image](https://github.com/ivanwilliammd/satusehat-integration/assets/28686631/84c0f83f-2658-4b95-928d-b09bcd784576)

Tetapi ketika saya vardump dan melihat log yang ada di database menunjukkan error 400 seperti ini
![image](https://github.com/ivanwilliammd/satusehat-integration/assets/28686631/0dc8fa61-e348-4ceb-b4c5-5e19e5156d9d)

ketika saya vardump result dari json nya dan saya masukin melalui postman itu 201 created, tetapi ketika di project saya muncul error seperti itu. Apakah code implementasi saya yang kurang tepat? Mohon bantuannya

**Comment by ivanwilliammd** [comment, marked as answer]
Salah UUID, yang digunakan harusnya UUID4
Kalau misalnya tidak bisa dengan library bawaan silakan https://github.com/ramsey/uuid

Selain itu best practicenya jangan lakukan $organization->json() dan coba lgsg $organization->post()

**Comment by fattakbar** [comment]
setelah mengganti UUID4, pesan errornya tetap sama dok. Btw apakah ini running di php 8.3.4 dok?
try {
            $uuid = Uuid::uuid4()->toString();
            $t = 'Nama Rumah Sakit';
        
            $organization = new Organization;
            $organization->addIdentifier($uuid); // unique string free text (increments / UUID / inisial)
            $organization->setName($t); // string free text
        
            [$statusCode, $response] = $organization->post();
        
            echo '<pre>' . var_export($response, true) . '</pre>';
        } catch (FHIRException $e) {
            echo "FHIRException: " . $e->getMessage();
        } catch (\Exception $e) {
            echo "Exception: " . $e->getMessage();
        }

**Comment by ivanwilliammd** [comment]
Sudah composer update dengan query yg paling baru?
Jadi tadi saya sdh explore, tergantung dari project dan cara calling-nya, library **guzzle** ini ada yg minta JSON object ada juga yg minta textual. Kalau saya merujuk pada dokumentasi baru akan mengirimkan json object, dan tested tdk ada masalah....

https://docs.guzzlephp.org/en/stable/quickstart.html#post-form-requests

**Comment by ivanwilliammd** [comment]
![image](https://github.com/ivanwilliammd/satusehat-integration/assets/48039956/0a12e642-3159-4b18-8e34-da51f9cc8f04)

Coba cek ke contoh implemementasinya agar seragam, di saya aman2 saja, dan terlampir SS dari github ini
 https://github.com/ivanwilliammd/satusehat-laravel-example
